cli/azd/cmd/infra_create.go

```diff
@@ -259,7 +259,31 @@ func (ica *infraCreateAction) Run(ctx context.Context, cmd *cobra.Command, args
 		return fmt.Errorf("deployment failed: %w", err)
 	}
 
+	dotnetCli := tools.NewDotNetCli()
```

Comment:
We should make two changes to checking for the dotnet CLI

1. Let's do it much earlier, before we do any of the infrastructure deployment.
2. We should only require a this is cases where there's at least one dotnet service in their project.

Probably the best way to do this is to move L262-L265 up to L80, and move this code into a loop that walks over proj.Services and if it finds one with a `.Language` of "dotnet" it calls tools.EnsureInstalled and if it succeeds breaks out of the loop.